docs/checkit.js

```diff
@@ -782,32 +778,33 @@ Calendar.prototype.addMonth = function() {
     
     // Update self.endDate so that it occurs one month later.
     self.endDate = self.endDate.add(1, 'months');
-    // Update the self.lastMonth property
-    self.lastMonth = new Month(self.endDate.format("YYYYMMDD"));
     // Update the endDateString in the calendarState
     self.state.endDateString = self.endDate.format("YYYYMMDD");
     // Update the self.monthObjects so that it includes the new month
-    self.monthObjects = self.generateMonthObjects();
+    self.monthObjects = self.generateMonthObjects(self.startDate, self.endDate);
```

Comment:
I was thinking something like..

```
var oldEndDate = Moment(self.endDate);
self.endDate.add(1, 'months');  // I think this just mutates the object instead of copying it and returning the new one right?
var newMonth = self.generateMonthObjects(oldEndDate, self.endDate);
self.monthObjects = self.monthObjects.push(newMonth);
return newMonth;
```

But regenerating all the month objects is probably fine. There aren't that many of them and it makes it harder to accidentally have stale state hanging around.